netlink-sys/src/socket.rs

```diff
@@ -252,34 +255,50 @@ impl Socket {
         // a pointer to it.
         let addrlen_ptr = &mut addrlen as *mut usize as *mut libc::socklen_t;
 
-        //                      Cast the *mut u8 into *mut void.
-        //               This is equivalent to casting a *char into *void
-        //                                 See [thread]
-        //                                       ^
-        //           Create a *mut u8            |
-        //                   ^                   |
-        //                   |                   |
-        //             +-----+-----+    +--------+-------+
-        //            /             \  /                  \
-        let buf_ptr = buf.as_mut_ptr() as *mut libc::c_void;
-        let buf_len = buf.len() as libc::size_t;
+        let chunk = buf.chunk_mut();
+        //                        Cast the *mut u8 into *mut void.
+        //                 This is equivalent to casting a *char into *void
+        //                                   See [thread]
+        //                                         ^
+        //             Create a *mut u8            |
+        //                    ^                    |
+        //                    |                    |
+        //             +------+-------+   +--------+-------+
+        //            /                \ /                  \
+        let buf_ptr = chunk.as_mut_ptr() as *mut libc::c_void;
+        let buf_len = chunk.len() as libc::size_t;
 
         let res = unsafe { libc::recvfrom(self.0, buf_ptr, buf_len, flags, addr_ptr, addrlen_ptr) };
         if res < 0 {
             return Err(Error::last_os_error());
+        } else {
+            let written = std::cmp::min(buf_len, res as usize);
```

Comment:
Interesting, I would have assumed `res <= buf_len` is always true. Did you add this "just in case", or is my assumption wrong?

EDIT: the answer if a couple lines below: my assumption is wrong.

Reply:
I'll simply copy the comment; also `MSG_TRUNC` is enough for this, `MSG_PEEK` is not needed afaict - I will change the comment accordingly.